### PR TITLE
Add spark-3.4.3 and 3.5.1 support

### LINKED ISSUE #381: [Feat][Spark] support of Spark 3.4.x and 3.5.x
**Is your feature request related to a problem? Please describe.**
Currently only 3.2.x and 3.3.x versions are supported.

**Describe the solution you'd like**
- `datasources-34` submodule in spark project
- `datasources-34` profile in top level pom.xml
- new version in spark CI `matrix`

**Describe alternatives you've considered**
.

**Additional context**
Existent `datasources-32` and `datasources-33` may be used as an example.

### PR DESCRIPTION
**Pull request by SemyonSinchenko**
### Reason for this PR
- support of Spark 3.4.x
- support of Spark 3.5.x
- update PySpark dependencies

### What changes are included in this PR?
- new datasource-34
- new datasource-35
- updates of pom files
- PySpark impl switched to Spark-3.5 as a most recent
- the default Python version is 3.10 at the moment
- Nebula Example is skipped for 3.4+ versions
- Spark codebase is prepared for scala 2.13

### Are these changes tested?
Unit tests

### Are there any user-facing changes?
Potentially switching the default Python version to 3.10 may affect the development setup of contributors. But there are no user-facing changes.

Close #381

### CODE REVIEW

**Comment by SemyonSinchenko** [comment]
It looks like it is still impossible to work with scala 2.13 from nebula. I will revert migration to 2.13

**Comment by acezen** [comment]
> It looks like it is still impossible to work with scala 2.13 from nebula. I will revert migration to 2.13



> It looks like it is still impossible to work with scala 2.13 from nebula. I will revert migration to 2.13

Thanks for the cool work!

**Review by Thespica — APPROVED** [review]
LGTM~

Thanks for your work!

File: .github/workflows/spark.yaml [hunk 1]
```diff
@@ -47,11 +47,17 @@ jobs:
       matrix:
         include:
           - mvn-profile: "datasources-32"
-            spark: "spark-3.2.2"
-            spark-hadoop: "spark-3.2.2-bin-hadoop3.2"
+            spark: "spark-3.2.4"
```
Comment (acezen) [inline .github/workflows/spark.yaml:50]: Hi, Sem, here I want to confirm that now we support 3.2.x or we just support the specific version 3.2.4? ditto to 3.3.x, 3.4.x and 3.5.x

**Review by acezen — APPROVED** [review]
LGTM

File: .github/workflows/spark.yaml (hunk 1 above)
Comment (SemyonSinchenko) [reply .github/workflows/spark.yaml:50]: We are supporting 3.2.x, but testing on the latest available (3.2.4)

**Pull request merged**